### PR TITLE
Moved To the Repository model spoken about and put clean Domain objects for the Front end to use.

### PR DESCRIPTION
**Pull request by CrypticEngima**

### CODE REVIEW

File: HacktoberfestProject.Web/Data/Entities/PrEntity.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿namespace HacktoberfestProject.Web.Models.Entities
+﻿namespace HacktoberfestProject.Web.Data.Entities
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Data/Entities/PrEntity.cs:1]: Why the move to Data from Models?

File: HacktoberfestProject.Web/Data/Entities/RepositoryEntity.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace HacktoberfestProject.Web.Data.Entities
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Data/Entities/RepositoryEntity.cs:4]: I prefer all models tobe in a `Models` directory for discoverability

File: HacktoberfestProject.Web/Data/Entities/UserEntity.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+using Microsoft.Azure.Cosmos.Table;
+using Microsoft.Azure.Documents;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using User = HacktoberfestProject.Web.Models.User;
+
+namespace HacktoberfestProject.Web.Data.Entities
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Data/Entities/UserEntity.cs:11]: As above

File: HacktoberfestProject.Web/Data/Repositories/UserRepository.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		public UserRepository(ITableContext tableContext)
+		{
+			_tableContext = tableContext ?? throw new ArgumentNullException(nameof(tableContext));
+		}
+
+		public async Task<User> CreateAsync(User user)
+		{
+			UserEntity userEntity = (UserEntity)user;
+
+			return (User)await _tableContext.InsertOrMergeEntityAsync(userEntity);
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Data/Repositories/UserRepository.cs:23]: I would have preferred these `await`s to be on a new line against a variable and return the variable for greater readability. I know it adds extra line of code but it's clearer for less experienced developers

File: HacktoberfestProject.Web/Models/Pr.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace HacktoberfestProject.Web.Models
```
Comment (Layla-P) [inline HacktoberfestProject.Web/Models/Pr.cs:6]: Is this so we don't pass the entity model to the view? I thought we decided that it was all public data anyway so we didn't need to create a DTO.

**Review by Layla-P — COMMENTED** [review]
I'm happy with everything except the move of the entity models to the data directory. I don't think we needed the DTO models as everything we have is public but we have them now :-)
I would like them to go in a Dto directory with in Models along with the entity models returning to  Models/Entities unless you have a strong objection in which case we can discuss further.

File: HacktoberfestProject.Web/Data/Entities/PrEntity.cs (hunk 1 above)
Comment (CrypticEngima) [reply HacktoberfestProject.Web/Data/Entities/PrEntity.cs:1]: Was purely to keep all database items together. I'll shift them to the models directory.

File: HacktoberfestProject.Web/Models/Pr.cs (hunk 5 above)
Comment (CrypticEngima) [reply HacktoberfestProject.Web/Models/Pr.cs:6]: The entity model has a load of extra fields from the TableEntity class it inherits from. I was trying to create a barrier that stopped this from happening so the models going to the controller/view only had relevant data.

**Review by Layla-P — APPROVED** [review]
Thanks for making the directory change :-)

**Pull request merged**